### PR TITLE
Pass options object to MSSQL removeColumn queries

### PR DESCRIPTION
**Pull request by harshithkashyap**
### Pull Request check-list
- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [ ] Does your issue contain a link to existing issue (Closes #[issue]) or a description of the issue you are solving?
- [ ] Have you added new tests to prevent regressions?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Have you added an entry under `Future` in the changelog?

### Description of change

Use lodash `_.assign` to pass user defined options such as transaction to MSSQL `removeColumn` queries

### CODE REVIEW

**Comment by harshithkashyap** [comment]
Ping @sushantdhiman @janmeier @felixfbecker

**Comment by sushantdhiman** [comment]
What issue this PR solves, looks like we need tests as well @harshithkashyap

File: lib/dialects/mssql/query-interface.js [hunk 1]
```diff
@@ -20,47 +22,47 @@
   @param  {Boolean|Function} [options.logging] A function that logs the sql queries, or false for explicitly not logging these queries
  @private
  */
-var removeColumn = function(tableName, attributeName, options) {
-  var self = this;
+const removeColumn = function(tableName, attributeName, options) {
   options = options || {};
+  options = _.assign({ raw: true, logging: options.logging}, options);
```
Comment (felixfbecker) [inline lib/dialects/mssql/query-interface.js]: Use `Object.assign`
`logging: options.logging` is redundant
Move the `|| {}` inside the call

**Comment by harshithkashyap** [comment]
@sushantdhiman  `sequelize.query` methods [here](https://github.com/sequelize/sequelize/blob/master/lib/dialects/mssql/query-interface.js#L28) does not pass the options param.
I've only modified the code to use lodash assign method and used es2015 wherever needed.

**Comment by sushantdhiman** [comment]
@harshithkashyap good, no tests needed then

File: changelog.md [hunk 2]
```diff
@@ -1,4 +1,5 @@
 # Future
+- [CHANGED] MSSQL `removeColumn` query to pass options to `sequelize.query` [#7139](https://github.com/sequelize/sequelize/pull/7193)
```
Comment (sushantdhiman) [inline changelog.md]: [FIXED] Properly pass options to `sequelize.query` in `removeColumn` [MSSQL] [#7139](https://github.com/sequelize/sequelize/pull/7193)

File: lib/dialects/mssql/query-interface.js [hunk 3]
```diff
@@ -24,7 +22,7 @@ const _ = require('lodash');
  */
 const removeColumn = function(tableName, attributeName, options) {
   options = options || {};
-  options = _.assign({ raw: true, logging: options.logging}, options);
+  options = Object.assign({}, options);
```
Comment (sushantdhiman) [inline lib/dialects/mssql/query-interface.js]: We need `raw: true` @harshithkashyap

Comment (sushantdhiman) [inline lib/dialects/mssql/query-interface.js]: It should be `Object.assign({ raw: true }, options || {} );`

File: lib/dialects/mssql/query-interface.js [hunk 4]
```diff
@@ -20,47 +20,47 @@
   @param  {Boolean|Function} [options.logging] A function that logs the sql queries, or false for explicitly not logging these queries
  @private
  */
-var removeColumn = function(tableName, attributeName, options) {
-  var self = this;
+const removeColumn = function(tableName, attributeName, options) {
   options = options || {};
+  options = Object.assign({}, options);
```
Comment (sushantdhiman) [inline lib/dialects/mssql/query-interface.js]: `Object.assign({ raw: true }, options || {});` , remove `options = options || {};`

File: lib/dialects/mssql/query-interface.js (hunk 3 above)
Comment (harshithkashyap) [inline lib/dialects/mssql/query-interface.js]: @sushantdhiman Would it not be set [here](https://github.com/sequelize/sequelize/blob/master/lib/sequelize.js#L493-L495)

File: changelog.md [hunk 5]
```diff
@@ -1,4 +1,5 @@
 # Future
+- [FIXED] Properly pass options to sequelize.query in removeColumn [MSSQL] [#7139](https://github.com/sequelize/sequelize/pull/7193)
```
Comment (sushantdhiman) [inline changelog.md]: Please highlight `sequelize.query` and `removeColumn` as code blocks ;)

Comment (harshithkashyap) [inline changelog.md]: Sorry, missed that :)

Comment (sushantdhiman) [inline changelog.md]: Also 7139 != 7193

Comment (harshithkashyap) [inline changelog.md]: @sushantdhiman Sorry about missing those. Thanks for catching. I've pushed with the suggested changes.

**Pull request merged**